### DISCUSSION TITLE
Documentation missing : any help needed ?

### DISCUSSION THREAD

**Discussion by TechWatching** [JoshClose/CsvHelper#1214] [Q&A]
I have started to use csv helper and the library is really great but I am sometimes losing time finding out how correctly configure or using the reader to handle some edge case.  Indeed some parts of the documentation are empty or "coming soon" .

- Do you have an estimated date for the release of the documentation at least the complete API documentation ?
- I have seen that there is a branch named docfx, are you planning to move to docfx as you primary tool for doc generation ?
- Do you need some help to write some of the markdown documentation ? I don't know much of the API as I started to use it recently for a little thing, but if needed I could try to contribute.

**Comment by JoshClose** [comment]
1. No. I haven't had much time lately to work on it. Once I get the base examples there, I'll just add things every time people ask a question.
1. I'm looking into DocFX. It does the api generation portion, and I believe it will merge in extra documentation on the api side, which would be really handy. I'm not sure how easy it is to accommodate the layout and stuff that I want. I've decided to finish up the api work before I play with DocFX any more. I'm nearing the end of writing an api doc generator that works with how I have things set up, so that shouldn't be too much longer.
1. Maybe at some point when my documentation process has settled. DocFX uses a different highlighter than I currently am, and I've customized the one I'm using to work with Bulma. I want the process of building the docs to be finalized beforehand.

For now, in Visual Studio you can go to `View -> Object Browser` and look at the api there. There aren't any examples in that though. I'm also thinking I want all examples to be external of the api, but I'm not set on that yet.

**Comment by TechWatching** [comment]
Ok, thanks for your answers. I did not think about "Object Browser" but it does the job perfectly, thanks for the tip :).

**Comment by mikeblas** [comment]
It remains true that no documentation is available. It's difficult to discover how to use the library without it. Is there any plan to make reference documentation again?

**Comment by AltruCoder** [comment]
The getting started page has a lot of documentation. https://joshclose.github.io/CsvHelper/getting-started/

**Comment by mikeblas** [comment]
The problem with "Getting Started" is that it is very limited in scope. (And it should be.) There is no API reference information at all, and anyone doing anything but the most trivial work will long for the information that it could provide.
